dashboard/src/modules/components/AuthComponent/signupFormData.js

```diff
@@ -0,0 +1,70 @@
+export const signupFormData = [
+  {
+    key: 1,
+    label: "First name",
+    id: "horizontal-form-first-name",
+    name: "firstName",
+    isRequired: true,
+    type: "text",
+  },
+  {
+    key: 2,
+    label: "Last name",
+    id: "horizontal-form-last-name",
+    name: "lastName",
+    isRequired: false,
+    type: "text",
+  },
+  {
+    key: 3,
+    label: "User name",
+    id: "horizontal-form-user-name",
+    name: "userName",
+    isRequired: true,
+    type: "text",
+  },
+  {
+    key: 4,
+    label: "Email address",
+    id: "horizontal-form-email-address",
+    name: "email",
+    isRequired: true,
+    type: "text",
+  },
+  {
+    key: 5,
+    label: "Password",
+    id: "horizontal-form-password",
+    name: "password",
+    isRequired: true,
+    type: "password",
+  },
+  {
+    key: 6,
+    label: "Confirm password",
+    id: "horizontal-form-confirm-password",
+    name: "passwordConfirm",
+    isRequired: true,
+    type: "password",
+  },
+];
+
+
+export const passwordConstraintsText = [
+  {
+    label: "8 characters",
+    name: "passwordLength",
+  },
+  {
+    label: "1 special character",
+    name: "passwordSpecialChars",
+  },
+  {
+    label: "1 number",
+    name: "passwordContainsNumber",
+  },
+  {
+    label: "1 uppercase letter",
+    name: "passwordBlockLetter",
+  }
```

Comment:
This generally seems to be compatible with the structure of Red Hat SSO; however instead of a fixed requirement, it allows "domains" to be structured with specific password policy settings, including a "number of character classes" and "minimum length". The "character classes" would include the special character, digits, uppercase, and lowercase; it doesn't appear you can require specific classes, but you can specify how many, with a default of 0. (That is, if you configure your domain with "2 classes", a valid password might have uppercase and lowercase, or digit and special character; we might need to have a server API for the dashboard to discover the confirmation set by the SSO domain.)

You're also hardcoding a minimum length of 8, and I know sites that require 12; again, Keycloak/Red Hat SSO allows setting that by domain, so eventually we'll need something more dynamic.

Right now, this is probably fine, although I've become a fan of the "random hyphenated words" password style we're not supporting here, which I suppose would qualify as a "2 character class" (hyphen and lowercase). Not worth worrying about right now, but the configuration is something to keep in mind.